packages/atomic/src/components/facets/facet-search/facet-search.tsx

```diff
@@ -1,53 +1,60 @@
+import {Component, h, Prop, State} from '@stencil/core';
 import {BaseFacetSearchResult} from '@coveo/headless/dist/api/search/facet-search/base/base-facet-search-response';
+import {
+  CategoryFacet,
+  CategoryFacetState,
+  Facet,
+  FacetState,
+} from '@coveo/headless';
+import {randomID} from '../../../utils/utils';
+import {I18nState} from '../../../utils/initialization-utils';
+import {CategoryFacetSearchResult} from '@coveo/headless/dist/api/search/facet-search/category-facet-search/category-facet-search-response';
 
 @Component({
   tag: 'facet-search',
+  styleUrl: 'facet-search.pcss',
   shadow: false,
 })
 export class FacetSearch {
-  @Event() public resultSelected!: EventEmitter<BaseFacetSearchResult>;
-  @Event() public showMoreResults!: EventEmitter<void>;
-  @Event() public facetSearch!: EventEmitter<string>;
-  @Prop() public facetSearchResults!: BaseFacetSearchResult[];
-  @Prop() public moreValuesAvailable!: boolean;
+  @Prop() public facet!: Facet | CategoryFacet;
+  @Prop() public facetState!: FacetState | CategoryFacetState;
+  @Prop({reflect: true, attribute: 'data-id'}) public _id = randomID(
+    'atomic-facet-search-'
+  );
 
-  private get facetSearchInput() {
-    return (
-      <input
-        onInput={(e) =>
-          this.facetSearch.emit((e.target as HTMLInputElement).value)
-        }
-        class="apply-border-on-background-variant w-full rounded"
-      />
-    );
-  }
-
-  private get facetSearchResultList() {
-    return this.facetSearchResults.map((searchResult) => (
-      <div onClick={() => this.resultSelected.emit(searchResult)}>
-        {searchResult.displayValue} {searchResult.count}
-      </div>
-    ));
-  }
-
-  private get showMoreSearchResults() {
-    if (!this.moreValuesAvailable) {
-      return null;
-    }
-
-    return (
-      <button onClick={() => this.showMoreResults.emit()}>show more</button>
-    );
-  }
+  @State()
+  public strings: I18nState = {
+    clear: () => 'clear',
+    search: () => 'search',
+    searchBox: () => 'searchbox',
+    querySuggestionList: () => 'querySuggestionList',
+  };
 
   public render() {
     return (
-      <div>
-        {this.facetSearchInput}
-        {this.facetSearchResultList}
-        {this.showMoreSearchResults}
-      </div>
+      <base-search
+        id={this._id}
+        moreValuesAvailable={this.facetState.facetSearch.moreValuesAvailable}
+        strings={this.strings}
+        hideSubmit={true}
+        suggestionValues={(this.facetState.facetSearch
+          .values as BaseFacetSearchResult[]).map(
+          (value: BaseFacetSearchResult) => ({
+            ...value,
+            value: `<span class="label">${value.rawValue}</span>
+          <span class="number-of-values">${value.count}</span>`,
+          })
+        )}
+        onSelectValue={(e: CustomEvent<number>) => {
+          const value = this.facetState.facetSearch.values[e.detail];
+          this.facet.facetSearch.select(value as CategoryFacetSearchResult);
```

Comment:
I couldn't find a way to avoid this cast and strictly speaking its not "correct". Obviously we know that value will be the correct type for the facetSearch's select method because they both originate from the same source. But the typescript compiler isn't aware of that and just sees two union types that can have incompatible combinations.